### PR TITLE
Static library build fixes.

### LINKED ISSUE #421: Static build fails for v2.4.0
The static build of v2.4.0 fails. I confirmed that a static build of v2.3.0 works OK for me.

```
$ git clone https://github.com/gnuradio/volk.git
$ cd volk/
$ git submodule update --init --recursive
$ git checkout v2.4.0
$ mkdir build && cd build
$ cmake -DENABLE_STATIC_LIBS=True ..
$ make


...
Scanning dependencies of target volk-config-info
[ 85%] Building CXX object apps/CMakeFiles/volk-config-info.dir/volk-config-info.cc.o
[ 86%] Building CXX object apps/CMakeFiles/volk-config-info.dir/volk_option_helpers.cc.o
[ 88%] Linking CXX executable volk-config-info
../lib/libvolk.a(volk_cpu.c.o): In function `i_can_has_avx512cd':
volk_cpu.c:(.text+0x28): undefined reference to `GetX86Info'
../lib/libvolk.a(volk_cpu.c.o): In function `i_can_has_avx512f':
volk_cpu.c:(.text+0x78): undefined reference to `GetX86Info'
../lib/libvolk.a(volk_cpu.c.o): In function `i_can_has_avx2':
volk_cpu.c:(.text+0xc8): undefined reference to `GetX86Info'
../lib/libvolk.a(volk_cpu.c.o): In function `i_can_has_avx':
volk_cpu.c:(.text+0x118): undefined reference to `GetX86Info'
../lib/libvolk.a(volk_cpu.c.o): In function `i_can_has_sse4_2':
volk_cpu.c:(.text+0x168): undefined reference to `GetX86Info'
../lib/libvolk.a(volk_cpu.c.o):volk_cpu.c:(.text+0x1a8): more undefined references to `GetX86Info' follow
collect2: error: ld returned 1 exit status
apps/CMakeFiles/volk-config-info.dir/build.make:121: recipe for target 'apps/volk-config-info' failed
make[2]: *** [apps/volk-config-info] Error 1
CMakeFiles/Makefile2:464: recipe for target 'apps/CMakeFiles/volk-config-info.dir/all' failed
make[1]: *** [apps/CMakeFiles/volk-config-info.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

### PR DESCRIPTION
**Pull request by drmpeg**
Fixes https://github.com/gnuradio/volk/issues/421

### CODE REVIEW

**Review by jdemel — CHANGES_REQUESTED** [review]
Thanks! LGTM. Could you double check formatting? That would be my change request. Besides, I'd like to merge this right after #420 .

**Comment by drmpeg** [comment]
Can you point to the formatting that you don't like? It's all copy and paste.

File: cmake/Modules/FindORC.cmake [hunk 1]
```diff
@@ -22,6 +22,10 @@ FIND_LIBRARY(ORC_LIB orc-0.4
              HINTS ${PC_ORC_LIBRARY_DIRS}
 	     PATHS ${ORC_ROOT}/lib${LIB_SUFFIX} ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
 
+FIND_LIBRARY(ORC_LIBRARY_STATIC liborc-0.4.a
+             HINTS ${PC_ORC_LIBRARY_DIRS}
+	     PATHS ${ORC_ROOT}/lib${LIB_SUFFIX} ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
```
Comment (jdemel) [inline cmake/Modules/FindORC.cmake:27]: I mainly stumbled over this section before I added my "please format" comment.

I actually found a nice tool to do auto formatting: https://github.com/cheshirekow/cmake_format
But that would change a lot of lines.

I just open a new issue to propose cmake formatting and we can leave this as it is for the moment.

**Review by jdemel — APPROVED** [review]

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -569,10 +569,11 @@ install(TARGETS volk
 #probably doesn't.
 if(ENABLE_STATIC_LIBS)
   add_library(volk_static STATIC $<TARGET_OBJECTS:volk_obj>)
-  target_link_libraries(volk_static PUBLIC ${volk_libraries} pthread)
+  target_link_libraries(volk_static PUBLIC ${volk_libraries} PRIVATE cpu_features)
```
Comment (michaelld) [inline lib/CMakeLists.txt]: With #420 merged, the `cpu_features` needs to be added within an `if(VOLK_CPU_FEATURES)` clause.

**Review by michaelld — CHANGES_REQUESTED** [review]
minor change & then I think this is good to go

File: lib/CMakeLists.txt (hunk 2 above)
Comment (drmpeg) [reply lib/CMakeLists.txt]: Updated.

**Review by michaelld — APPROVED** [review]
LGTM; thx!

**Comment by michaelld** [comment]
@jdemel do you want a quick re-review? If you're good here then go ahead & merge.

**Review by jdemel — APPROVED** [review]
LGTM

**Pull request merged**